Tests/MongoSwiftTests/DocumentTests.swift

```diff
@@ -680,4 +682,114 @@ final class DocumentTests: MongoSwiftTestCase {
         expect(doc["autoclosure test", default: "\(stringVal) and \(floatVal)" + stringVal])
             .to(bsonEqual("\(stringVal) and \(floatVal)" + stringVal))
     }
+
+    func testUUIDCodingStrategies() throws {
+        // TODO: add encoding strategy tests SWIFT-282
+
+        // swiftlint:disable nesting
+        struct UUIDWrapper: Codable {
+            let uuid: UUID
+        }
+        // swiftlint:enable nesting
+
+        // randomly generated uuid
+        guard let uuid = UUID(uuidString: "2c380a6c-7bc5-48cb-84a2-b26777a72276") else {
+            throw MongoError.bsonDecodeError(message: "Cant create URI")
+        }
+
+        let decoder = BSONDecoder()
+
+        // UUID default decoder expects a string
+        decoder.uuidDecodingStrategy = .deferredToUUID
```

Comment:
It works fine with and without `@testable`. `open` really only matters when it comes to inheritance and overriding, not simply assigning (from what I understand).

Reply:
huh, interesting, I wonder why they bothered to make `JSONDecoder` open then....

Reply:
thinking about this more.... my guess is that they did it that way so you could subclass `JSONDecoder` and override properties on it for the entirety of your application rather than having to set it every time you create a new decoder. seems potentially valuable. thoughts?

Reply:
I'm not really sure. There isn't a _ton_ of value in subclassing `BSONEncoder` because it doesn't really do anything by itself; it's just a wrapper of the real encoder. The only real reason you'd want to do it would be to, as you said, override the default properties, but you don't need a subclass for that (e.g. define a `makeDecoder` factory function which configures it for you).

Reply:
Good point. Let’s leave it then, can always add later if someone has a new compelling reason.